python/tester/upload_hsc_rc2.py

```diff
@@ -102,14 +102,17 @@ def get_hsc_visit_list(butler, n_sample):
     visits : `list` [`int`]
         A list of randomly selected visit IDs from the HSC-RC2 dataset
     """
+    if n_sample > 432:
+        raise ValueError(f"Requested {n_sample} groups, but only 432 are available.")
+
```

Comment:
I'm not sure this number needs to be hardcoded; what if the collection gets bigger (more data!) or smaller (rejected or failed processing)? I'd suggest either catching the `ValueError` raised by `random.sample`, or simply passing it through.

Reply:
This number is going to stay the same unless we have to a new dataset (e.g. RC3)  or when we change the implementation (e.g. allowing reuse of the same visit for different group). Because we are drawing the raw data, failed DRP processing doesn't matter either; DRP processing provides templates only. Nonetheless,  I agree it's better not to hardcode it. I feel it's nicer if the error message tells the user how many groups are available so the user can adjust for the next test.